sftp/location.go

```diff
@@ -0,0 +1,166 @@
+package sftp
+
+import (
+	"net/url"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/graymeta/stow"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"github.com/pkg/sftp"
+	"golang.org/x/crypto/ssh"
+)
+
+type location struct {
+	// we keep config here so that we can access the server information (username/host)
+	// when constructing the item urls.
+	config     *conf
+	sshClient  *ssh.Client
+	sftpClient *sftp.Client
+}
+
+// CreateContainer creates a new container, in this case a directory on the remote server.
+func (l *location) CreateContainer(containerName string) (stow.Container, error) {
+	if err := l.sftpClient.Mkdir(containerName); err != nil {
+		return nil, err
+	}
+
+	return &container{
+		location: l,
+		name:     containerName,
+	}, nil
+}
+
+// Containers returns a slice of the Container interface, a cursor, and an error.
+func (l *location) Containers(prefix, cursor string, count int) ([]stow.Container, string, error) {
+	infos, err := l.sftpClient.ReadDir(".")
+	if err != nil {
+		return nil, "", err
+	}
+
+	sort.Slice(infos, func(i, j int) bool { return infos[i].Name() < infos[j].Name() })
+
+	var cont []stow.Container
+	for i, v := range infos {
+		if !v.IsDir() {
+			continue
+		}
+
+		if !strings.HasPrefix(v.Name(), prefix) {
+			continue
+		}
+
+		if v.Name() <= cursor {
```

Comment:
how does this cursor work? if the name is lesser ascii val than the cursor we break skip it. Does this need to be case insensitive to account for different OS's?

Reply:
i guess it doesn't matter since its sorted already 🤷‍♂